packages/messaging-amqp/src/AmqpModule.ts

```diff
@@ -0,0 +1,417 @@
+/*
+ * © Copyright 2022 HP Development Company, L.P.
+ * SPDX-License-Identifier: MIT
+ */
+
+import {
+	App,
+	executeInterceptorsStack,
+	getInterceptorsHandlers,
+	InterceptorBag,
+	mapParallel,
+	mapSeries,
+	Module,
+	nextTick
+} from '@davinci/core';
+import { ClassReflection, ClassType, DecoratorId, MethodReflection, PartialDeep } from '@davinci/reflector';
+import { Level, Logger, pino } from 'pino';
+import createDeepmerge from '@fastify/deepmerge';
+import amqplib, { Channel } from 'amqplib';
+import amqpConnectionManager, { AmqpConnectionManager, ChannelWrapper } from 'amqp-connection-manager';
+import stringify from 'fast-json-stable-stringify';
+import { EventEmitter } from 'events';
+import { ParameterConfiguration, Subscription, SubscriptionSettings } from './types';
+import { SubscribeDecoratorMetadata } from './decorators/types';
+
+const deepmerge = createDeepmerge({ all: true });
+
+export interface AmqpModuleOptions {
+	connection: string | amqplib.Options.Connect;
+	connectionTimeout?: number;
+	subscriptions?: Array<SubscriptionSettings>;
+	defaultSubscriptionSettings?: PartialDeep<SubscriptionSettings>;
+	gracefulShutdownStrategy?: 'none' | 'processInFlight' | 'nackInFlight' | null;
+	logger?: {
+		name?: string;
+		level?: Level | 'silent';
+	};
+}
+
+export class AmqpModule extends Module {
+	app: App;
+	private logger: Logger;
+	private options: AmqpModuleOptions;
+	private connection: AmqpConnectionManager;
+	private subscriptions: Array<Subscription>;
+	private subscriptionHash: Record<string, Subscription> = {};
```

Comment:
Should we move to use `Map` instead of `Object` for these dictionaries?

Reply:
👍